config/tls.go

```diff
@@ -58,31 +75,37 @@ func (t *TLS) MakeConfig(serverName string) (*tls.Config, error) {
 	}
 
 	tlsConfig := &tls.Config{MinVersion: tls.VersionTLS12}
-	if t.Cert == "" {
-		if t.Key != "" {
-			return nil, errors.New("private key given, but client certificate missing")
-		}
-	} else if t.Key == "" {
+	if t.Cert == "" && t.Key != "" {
```

Comment:
I would omit the else statements for the early exit conditions:

```golang
if (key-only) {
    return
}

if (cert-only) {
    return
}

if (cert-and-key) {
    ...
}
```

Reply:
Please take another look.